### DISCUSSION TITLE
Becoming a committer

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#759] [Q&A] [answered]
I'm investing heavily in Serverpod, and I have been submitting a bunch of pull requests to fix bugs, as well as a major feature addition (Facebook login). What does it take to become a committer for this project?

**Comment by vlidholt** [comment, marked as answer]
As soon as you have an approved PR your name will be included among contributors (it may take a day or so for it to appear).

**Reply by lukehutch** [reply]
OK, thanks, but it's not credit I'm looking for, I just want to move Serverpod forward, because there are features I need :-) What is the process for getting PRs reviewed and committed to main, for this project? I'm juggling so many open PRs right now that it's starting to get really tricky to keep working on serverpod, because often the PRs are interrelated, and I often need one PR to be committed before I can rebase off of main to create the next PR.

**Reply by vlidholt** [reply]
That is awesome! We are usually much faster on reviewing PRs. I've in the middle of recruiting a few more full time employees, plus I'm selling my house. 😅

To make the process smoother, make sure that there are tests for all new code additions and that all the existing tests pass. You can run the tests either locally or if you clone the Github repo, you can run them when you push to your own repo.

**Comment by lukehutch** [comment]
I didn't have a clue how to test any of the things I submitted PRs for, other than observing that it fixed the problem for me.

When it came to adding Facebook login, I copied the Google login code, which doesn't even have any tests. Plus testing Facebook login would be difficult, because it would require creating Facebook OAuth2 login credentials just for the test, and those would be linked to my personal Facebook account.

**Reply by vlidholt** [reply]
Yeah, for the sign in with Facebook, that is a tough one to test, so I will do that manually. It was more meant like a general note. We are improving test coverage overall, so (most) new code need to have tests. Especially important as more people get involved in the project. :)
